docs/src/test/scala/sample/scaladsl/ConsumerExample.scala

```diff
@@ -341,3 +341,29 @@ object ExternallyControlledKafkaConsumer extends ConsumerExample {
   }
 }
 
+//Restart Kafka consumer
+object RestartKafkaConsumeExample extends ConsumerExample {
+  def main(args: Array[String]): Unit = {
+    // #restartConsumer
+    val runnableGraph: RunnableGraph[Consumer.Control] =
+      Consumer.committableSource(consumerSettings, Subscriptions.topics("topic1"))
+        .map(msg => { println(msg.record.value()); msg.committableOffset })
+        .map(_.commitScaladsl())
+        .toMat(Sink.ignore)(Keep.left)
+    val control1: Consumer.Control = runnableGraph.run()
+    val shutdownFuture = control1.shutdown()
+
+    shutdownFuture onComplete {
+      case Success(Done) =>
+        println("Stream was shutdown")
+
+        // Start the stream again
+        val control2: Consumer.Control = runnableGraph.run()
```

Comment:
@kciesielski  Yes the example is not realistic.  How about an actor with two states (Processing and NotProcessing).  In the NotProcessing state when receiving a Start message the state changes to Processing and the stream is started.  Conversely in the Processing state, receiving the Stop message results in stopping the stream and transitioning to the NotProcessing state.  

The two states could represent whether or not a downstream dependency is available to receive data.